content/post/language_migrations.md

```diff
@@ -0,0 +1,512 @@
+
+Although there are ten times more lines of code on GitHub in **PHP** than in **Ruby**, we have the same probability to switch for those two.
+
+**Go** (3.2 %) appears on 9th position which is largely honorable given the small proportion (0.9 %) of **Go** projects which are written on GitHub. The same proportion of projects are run in **Perl** for example, but this language doesn't really awake passion (2 % popularity).
+
+### What about sticking to a language ?
+
+If we keep the main diagonal of our transition matrix before applying the power iteration, we obtain slightly different results. It mainly decreases top languages popularity while raising smaller ones'. Indeed, it seems logical to believe that developers who invest their time in mastering a language on the fringe of the others tend to stick to them unlike popular ones.
+
+In the rest of the post, we will consider our first representation of the dominant eigenvector we gave.
+
+### Back to the transition matrix
+
+What if we sort our transition matrix so that the most popular languages appear at the bottom, and compare it to Erik's.
+
+<div class="grid2x">
+<div>
+<div>
+<img src="/post/language_migrations/sum_matrix_22lang_eig.svg">
+<p align="center" class="dt"><pre><code class="hljs python">source{d}'s sorted transition matrix</code></pre></p>
+</div>
+</div>
+<div>
+<div>
+<img src="/post/language_migrations/erik_green.png">
+<p align="center" class="dt"><pre><code class="hljs python">Erik's sorted transition matrix</code></pre></p>
+</div>
+</div>
+</div>
+
+* Developers coding in one of the 5 most popular languages (**Java**, **C**, **C++**, **PHP**, **Ruby**) are more likely to switch to **Python** with around that 22 % of chance in average.
+
+* Also according to Erik's matrix, in case we use **Ojective-C**, we would switch to **Swift** and reversely with the greatest probability (24 and 19 %).
+
+* Similarly, a **Visual Basic** developer has more chance (24 %) to move to **C#**, when Erik's reaches the nearly certitude with 92 %.
+
+* GitHub users of **Clojure**, **C#** and above all **Scala** would rather switch to **Java** with respectively 22, 29 and 40 % of chance which is the highest probability.
+
+* Most of people using numerical and statistical environments such as **Fortran** (36 %), **Matlab** (33 %) or **R** (40 %), move to **Python** unlike Erik's matrix which predicts **C** as their future language.
+
+* One common point I found with Erik's results about **golang** is that relatively to the previous language, the highest proportion of people moving to **Go**, belongs to people giving up with **Rust** (9 % for our matrix and 47 % for Erik's).
+
+## Over the last 16 years
+
+Now, focusing on a specific year, what do the yearly transition matrices look like ? Do they express the same find of language profile ? How has it been evolving since the early 20's ? Here is a sample of 4 matrices to feel the time-line.
```

Comment:
Now, we consider specific years and determine how the yearly transition matrices look like. <missing reference to the top, where you promised this>

The early 2000's. <I was surprised, people coding in 1920's, hehe>

sample of 4 matrices from various timeline intervals: